tests/cose.rs

```diff
@@ -149,22 +149,22 @@ fn test_de_order<T: Serialize + DeserializeOwned + Debug + PartialEq>(data: T) -
 fn de_p256() {
     let x = Bytes::from_slice(&[0xff; 32]).unwrap();
     let y = Bytes::from_slice(&[0xff; 32]).unwrap();
-    let key = P256PublicKey { x, y };
+    let key = PublicKey::P256Key(P256PublicKey { x, y });
     test_de("a5010203262001215820ffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff225820ffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff", key);
```

Comment:
Please keep the test for the direct deserialization, e. g.:
```suggestion
    let key = P256PublicKey { x, y };
    let data = "a5010203262001215820ffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff225820ffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff";
    test_de(data, key.clone());
    test_de(data, PublicKey::P256Key(key));
```